File: user-service/src/main/java/rs/raf/user_service/repository/AuthTokenRepository.java
package rs.raf.user_service.repository;

import org.springframework.data.jpa.repository.JpaRepository;
import rs.raf.user_service.domain.entity.AuthToken;

import java.util.Optional;

public interface AuthTokenRepository extends JpaRepository<AuthToken, Long> {
    Optional<AuthToken> findByToken(String token);

}

File: user-service-kotlin/src/main/kotlin/pack/userservicekotlin/repository/AuthTokenRepository.kt
package pack.userservicekotlin.repository

import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.stereotype.Repository
import pack.userservicekotlin.domain.entities.AuthToken
import java.util.*

@Repository
interface AuthTokenRepository : JpaRepository<AuthToken, Long> {
    fun findByToken(token: String): Optional<AuthToken>
}
